File: Beecrowd/4-Estruturas_e_Bibliotecas/1424-Problema_Fácil_de_Rujia_Liu-8/1424.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        string line;
        while ((line = Console.ReadLine()) != null) {
            string[] parts = line.Split();
            int n = int.Parse(parts[0]), m = int.Parse(parts[1]);

            Dictionary<int, List<int>> occurrence = new Dictionary<int, List<int>>();
            string[] array = Console.ReadLine().Split();
            
            for (int i = 0; i < n; i++) {
                int num = int.Parse(array[i]);
                if (!occurrence.ContainsKey(num)) {
                    occurrence[num] = new List<int>();
                }
                occurrence[num].Add(i + 1);
            }

            for (int i = 0; i < m; i++) {
                string[] query = Console.ReadLine().Split();
                int init = int.Parse(query[0]), wanted = int.Parse(query[1]);

                if (occurrence.ContainsKey(wanted) && occurrence[wanted].Count >= init) {
                    Console.WriteLine(occurrence[wanted][init - 1]);
                } else {
                    Console.WriteLine(0);
                }
            }
        }
    }
}

File: Beecrowd/4-Estruturas_e_Bibliotecas/1424-Problema_Fácil_de_Rujia_Liu-8/1424.py
import sys

for linha in sys.stdin:
    ocurrence = {}
    n, m = map(int, linha.split()) 
    array = list(map(int, input().split())) 
    idx=1
    for i in array:
        if not i in ocurrence:
            ocurrence[i] = []
        ocurrence[i].append(idx)
        idx+=1
    for c in range(m):
        init, wanted = map(int, input().split( ))
        print(0 if not wanted in ocurrence or len(ocurrence[wanted])<init else ocurrence[wanted][init-1])
        